datadog/fwprovider/resource_datadog_notification_rule.go

```diff
@@ -0,0 +1,401 @@
+	}
+	if !plan.Selectors.Severities.IsNull() {
+		severities := make([]datadogV2.RuleSeverity, len(plan.Selectors.Severities.Elements()))
+		resp.Diagnostics.Append(plan.Selectors.Severities.ElementsAs(ctx, &severities, false)...)
+		selectors.SetSeverities(severities)
+	}
+	if !plan.Selectors.Query.IsNull() {
+		selectors.SetQuery(plan.Selectors.Query.ValueString())
+	}
+
+	targets := make([]string, len(plan.Targets.Elements()))
+	resp.Diagnostics.Append(plan.Targets.ElementsAs(ctx, &targets, false)...)
+
+	ruleAttributes := datadogV2.NewCreateNotificationRuleParametersDataAttributesWithDefaults()
+	ruleAttributes.SetName(plan.Name.ValueString())
+	ruleAttributes.SetSelectors(*selectors)
+	ruleAttributes.SetTargets(targets)
+	if !plan.Enabled.IsNull() {
+		ruleAttributes.SetEnabled(plan.Enabled.ValueBool())
+	}
+	if !plan.TimeAggregation.IsNull() {
+		ruleAttributes.SetTimeAggregation(plan.TimeAggregation.ValueInt64())
+	}
+
+	body := datadogV2.NewCreateNotificationRuleParametersWithDefaults()
+	body.Data = datadogV2.NewCreateNotificationRuleParametersData(*ruleAttributes, datadogV2.NOTIFICATIONRULESTYPE_NOTIFICATION_RULES)
+
+	writeMutex.Lock()
+	defer writeMutex.Unlock()
+
+	var response datadogV2.NotificationRuleResponse
+	if *triggerSource == datadogV2.TRIGGERSOURCE_SECURITY_SIGNALS {
+		response, _, err = r.api.CreateSignalNotificationRule(r.auth, *body)
+	} else {
+		response, _, err = r.api.CreateVulnerabilityNotificationRule(r.auth, *body)
+	}
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating notification rule", err.Error())
+		return
+	}
+	if err := utils.CheckForUnparsed(response); err != nil {
+		resp.Diagnostics.AddError("response contains unparsed object", err.Error())
+		return
+	}
+
+	r.updateStateFromResponse(ctx, &plan, &response)
+	resp.Diagnostics.Append(resp.State.Set(ctx, plan)...)
+}
+
+func (r *NotificationRuleResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state notificationRuleModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	triggerSource, err := datadogV2.NewTriggerSourceFromValue(state.Selectors.TriggerSource.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Invalid trigger source", err.Error())
+		return
+	}
+
+	var response datadogV2.NotificationRuleResponse
+	if *triggerSource == datadogV2.TRIGGERSOURCE_SECURITY_SIGNALS {
+		response, _, err = r.api.GetSignalNotificationRule(r.auth, state.ID.ValueString())
+	} else {
+		response, _, err = r.api.GetVulnerabilityNotificationRule(r.auth, state.ID.ValueString())
+	}
+	if err != nil {
+		resp.Diagnostics.AddError("Error reading notification rule", err.Error())
+		return
+	}
+
+	r.updateStateFromResponse(ctx, &state, &response)
+	resp.Diagnostics.Append(resp.State.Set(ctx, state)...)
+}
+
+func (r *NotificationRuleResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
```

Comment:
nit: Update seems pretty similar to create, can we group them to avoid code dup?